### PR TITLE
Misc: Always try at least 1 block; raise original Exception in processify; bump certifi; ignore empty windows

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- pixetl may estimate that the most data it can operate on at a time is 0 blocks
- Exceptions raised by a function wrapped in processify that take more than one argument will not correctly be propagated
- The version of certifi package installed has known security vulnerabilities
- We fail in cases where we generate empty windows to operate on, which seems harmless to me

Issue Number: N/A

## What is the new behavior?

- pixetl will always try to operate on at least 1 block
- processify raises the original exception object, rather than attempting to re-create it by creating a new one
- certifi is updated
- empty windows are ignored

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: gfw_pixetl/decorators.py [hunk 1]
```diff
@@ -79,9 +79,7 @@ def wrapper(*args, **kwargs):
             raise SubprocessKilledError("Process was killed")
         elif error:
             ex_type, ex_value, tb_str = error
```
Comment (gtempus) [inline gfw_pixetl/decorators.py:81]: When unpacking unused data is a `_` a good idea? Just wondering why you kept the unused variables.
```suggestion
            _, ex_value, _ = error
```

File: gfw_pixetl/tiles/raster_src_tile.py [hunk 2]
```diff
@@ -373,15 +373,25 @@ def _windows(self, dst: DatasetWriter) -> Iterator[Window]:
...
                         # Figure out what it means to fail, and fix the window
                         # generating code in this function
                         LOGGER.warning(
                             f"Bogus window generated for tile {self.tile_id}! "
                             f"i: {i} j: {j} max_i: {max_i} max_j: {max_j} window: {window}"
                         )
-                    elif not (str(e) == "windows do not intersect"):
+                    elif "Intersection is empty Window" in e_str:
+                        # Seems harmless to skip empty windows we generate
+                        continue
```
Comment (gtempus) [inline gfw_pixetl/tiles/raster_src_tile.py:387]: Thanks for the comment! 🙌

**Review by gtempus — CHANGES_REQUESTED** [review]
Thanks, @dmannarino! I just had a question about unpacking data into unused variables.

File: gfw_pixetl/decorators.py (hunk 1 above)
Comment (dmannarino) [reply gfw_pixetl/decorators.py:81]: Good point, thanks!

**Review by gtempus — APPROVED** [review]

**Pull request merged**